plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+
+import {
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { zipObject } from 'lodash';
+import { createTemplateAction } from '../../createTemplateAction';
+import { Octokit } from '@octokit/rest';
+import { OctokitResponse } from '@octokit/types';
+import { InputError } from '@backstage/errors';
+import { createPullRequest } from 'octokit-plugin-create-pull-request';
+import globby from 'globby';
+
+const readFile = promisify(fs.readFile);
+
+type PullRequestResponse = {
+  html_url: string;
+};
+
+export interface PullRequestCreator {
+  createPullRequest(
+    options: createPullRequest.Options,
+  ): Promise<OctokitResponse<PullRequestResponse> | null>;
+}
+
+export type GithubPullRequestActionInput = {
+  owner: string;
+  repo: string;
+  title: string;
+  branchName: string;
+  description: string;
+  host?: string;
+  targetPath?: string;
+  sourcePath?: string;
+};
+
+export type ClientFactoryInput = {
+  host: string;
+  owner: string;
+  repo: string;
+};
+
+export const createClientFactory = async (
+  integrations: ScmIntegrationRegistry,
+  { owner, repo, host = 'github.com' }: ClientFactoryInput,
+): Promise<Octokit> => {
+  const integrationConfig = integrations.github
```

Comment:
think you can do `integrations.byUrl(...)` instead :)

Reply:
and in this case you want the github specific ones (with the right types), which also is possible with `integrations.github.byUrl` (or, in this case you may want `integrations.github.byHost`)